workflow-controller/controller-service/src/main/java/Converters/Converter.java

```diff
@@ -0,0 +1,123 @@
+     * @param inputFilePath
+     * @return assigns the file to field if valid.
+     * @throws Exception contains wrapped HyscaleExcpetion for Handling.
+     */
+    @Override
+    public List<File> convert(String inputFilePath) throws Exception {
+        ProcessingReport processingReport;
+        List<File> inputFiles = new ArrayList<File>();
+
+        if (StringUtils.isBlank(inputFilePath)) {
+            WorkflowLogger.error(ControllerActivity.EMPTY_FILE_PATH);
+            throw new HyscaleException(CommonErrorCode.EMPTY_FILE_PATH);
+        }
+
+        File file = new File(inputFilePath);
+        if (!file.exists()) {
+            WorkflowLogger.error(ControllerActivity.CANNOT_FIND_FILE, inputFilePath);
+            throw new HyscaleException(CommonErrorCode.FILE_NOT_FOUND);
+        }
+        if (file.isFile()) {
+            String fileName = file.getName();
+            if (!fileName.matches(getFilePattern())) {
+                logger.warn(getWarnMessage(), fileName);
+            }
+            if (file.length() == 0) {
+                WorkflowLogger.error(ControllerActivity.EMPTY_FILE_FOUND, inputFilePath);
+                throw new HyscaleException(CommonErrorCode.EMPTY_FILE_FOUND, inputFilePath);
+            }
+            try {
+                validateData(file);
+                processingReport = SchemaProcessor.validateSpec(file, getReferenceSchema(), ServiceInputType.YAML);
+            } catch (HyscaleException e) {
+                WorkflowLogger.error(ServiceSpecActivity.ERROR, e.getMessage());
+                throw e;
+            }
+            if (processingReport.isSuccess()) {
+                inputFiles.add(file);
+                return inputFiles;
+            }
+            StringBuilder messageBuilder = new StringBuilder();
+            StringBuilder exceptionBuilder = new StringBuilder();
+            Iterator<ProcessingMessage> messageIterator = processingReport.iterator();
+            while (messageIterator.hasNext()) {
+                ProcessingMessage message = messageIterator.next();
+                messageBuilder.append(message.getMessage());
+                exceptionBuilder.append(message.asException().fillInStackTrace());
+            }
+            WorkflowLogger.error(ServiceSpecActivity.ERROR, messageBuilder.toString());
+            logger.error(messageBuilder.toString());
+            logger.error(exceptionBuilder.toString());
+            throw new HyscaleException(ServiceSpecErrorCodes.INVALID_FORMAT, messageBuilder.toString());
+        }
+        if (file.isDirectory()) {
+            WorkflowLogger.error(ControllerActivity.DIRECTORY_INPUT_FOUND, inputFilePath);
+            throw new HyscaleException(CommonErrorCode.FOUND_DIRECTORY_INSTEAD_OF_FILE, inputFilePath);
+        }
+        WorkflowLogger.error(ControllerActivity.UNABLE_TO_PROCESS_INPUT, inputFilePath);
+        throw new HyscaleException(CommonErrorCode.CANNOT_PROCESS_INPUT);
+    }
+
```

Comment:
Upfront validation, if we don't know why service spec failed, how will user know ? So make all possible validations upfront and return.